applications/CompressiblePotentialFlowApplication/tests/naca0012_small_transonic_test/naca0012_small_transonic_parameters.json

```diff
@@ -42,10 +42,9 @@
                 "heat_capacity_ratio": 1.4,
                 "speed_of_sound": 340.3,
                 "inlet_potential": 1.0,
-                "mach_number_limit": 0.94,
                 "critical_mach": 0.90,
                 "upwind_factor_constant": 1.0,
-                "mach_number_squared_limit":3.0,
+                "mach_number_limit":1.73205080756887729353,
```

Comment:
@marcnunezc don't you find this a bit cumbersome for the user? Do you have any suggestion on how we can improve this?

By the way, we still have another parameter called critical_mach :sweat_smile:  This one is only used in the transonic element but it also indicates the onset of supersonic flow (as in the subsonic element). Another idea would be to remove mach_number_limit and keep critical_mach and mach_number_squared_limit, but we would need to see how to implement this so that it works for both subsonic and transonic elements

Reply:
> @marcnunezc don't you find this a bit cumbersome for the user? Do you have any suggestion on how we can improve this?
 
One thing that Im thinking now is that we can consider to enable both again, but throw a warning if both are defined at the same time (prioritizing one or the other). Thus, the user can:

- Define `mach_limit`, which will set `MACH_LIMIT=mach_limit` and ` MACH_LIMIT_SQUARED=mach_limit**2`
- Or define `mach_limit_squared` which will set   ` MACH_LIMIT_SQUARED=mach_limit_squared` and `MACH_LIMIT=sqrt(mach_limit_squared)`
- If both are defined, throw warning saying that the two parameters are defined and that only one will be used. 

This will avoid having two independant parameters that should be related (as we have now), while allowing to use one or the other depending on what is better for the user.  In terms of implementation,  it _shouldnt_  matter if we use one or the other as we would make sure both are related.

> By the way, we still have another parameter called critical_mach sweat_smile This one is only used in the transonic element but it also indicates the onset of supersonic flow (as in the subsonic element). Another idea would be to remove mach_number_limit and keep critical_mach and mach_number_squared_limit, but we would need to see how to implement this so that it works for both subsonic and transonic elements

The problem here is that conceptually `critical_mach` and `mach_limit` are not the same right? I understand one is the `free_stream_critical_mach` and the other the `local_mach_limit`

Reply:
@marcnunezc I tried to address your suggestion in the last commit

Reply:
Cool!